From 2b7b661810440179264e91cdb5b5323eb897d243 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:08 -0400
Subject: [tools] perf/trace: Handle MSG_WAITFORONE not defined

Message-id: <1408462094-14194-140-git-send-email-jolsa@redhat.com>
Patchwork-id: 88095
O-Subject: [PATCH RHEL7.1 BZ1131394 139/325] perf trace: Handle MSG_WAITFORONE not defined
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit b6e8f8f4674be5a32f78027ec6e432f5ea33921e
Author: David Ahern <dsahern@gmail.com>
Date: Sun Sep 22 19:44:56 2013 -0600

description
===========
Needed for compiles on Fedora 12 for example.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index c993a8b..24b9370 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -363,6 +363,9 @@ static size_t syscall_arg__scnprintf_socket_type(char *bf, size_t size,
 #ifndef MSG_PROBE
 #define MSG_PROBE      0x10
 #endif
+#ifndef MSG_WAITFORONE
+#define MSG_WAITFORONE 0x10000
+#endif
 #ifndef MSG_SENDPAGE_NOTLAST
 #define MSG_SENDPAGE_NOTLAST 0x20000
 #endif
-- 
1.7.1